File: frontend/src/services/attendanceService.js
import api from './api';

export const attendanceService = {
  getAttendance: (params) => api.get('/attendance', { params }),
  getMyAttendance: (params) => api.get('/attendance/my-attendance', { params }),
  getAttendanceById: (id) => api.get(`/attendance/${id}`),
  createAttendance: (data) => api.post('/attendance', data),
  updateAttendance: (id, data) => api.put(`/attendance/${id}`, data),
  deleteAttendance: (id) => api.delete(`/attendance/${id}`),
  checkIn: () => api.post('/attendance/check-in'),
  checkOut: () => api.post('/attendance/check-out'),
};

export default attendanceService;

File: backend/src/main/java/com/workpulse/ems/service/AttendanceService.java
package com.workpulse.ems.service;

import com.workpulse.ems.dto.request.AttendanceRequest;
import com.workpulse.ems.dto.response.AttendanceResponse;
import com.workpulse.ems.entity.enums.AttendanceStatus;
import com.workpulse.ems.security.services.UserDetailsImpl;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;

public interface AttendanceService {

    Page<AttendanceResponse> getAttendanceRecords(
            Long employeeId,
            LocalDate date,
            LocalDate startDate,
            LocalDate endDate,
            AttendanceStatus status,
            Pageable pageable
    );

    AttendanceResponse getAttendanceById(Long id, UserDetailsImpl userDetails);

    Page<AttendanceResponse> getMyAttendance(UserDetailsImpl userDetails, Pageable pageable);

    AttendanceResponse createAttendance(AttendanceRequest request);

    AttendanceResponse updateAttendance(Long id, AttendanceRequest request);

    void deleteAttendance(Long id);

    AttendanceResponse checkIn(UserDetailsImpl userDetails);

    AttendanceResponse checkOut(UserDetailsImpl userDetails);
}
